hazelcast/src/main/java/com/hazelcast/sql/impl/QueryUtils.java

```diff
@@ -71,6 +73,13 @@ public static HazelcastSqlException toPublicException(Throwable e, @Nonnull UUID
 
             return new HazelcastSqlException(originatingMemberId, e0.getCode(), e0.getMessage(), e, e0.getSuggestion());
         } else {
+            while (copy != null) {
```

Comment:
minor: we can move the definition of `Throwable copy = e;` here, where it is needed

Reply:
done